receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -0,0 +1,88 @@
+package k8swindows
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
+	"go.uber.org/zap"
+	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+)
+
+type kubeletSummaryProvider struct {
+	logger   *zap.Logger
+	hostIP   string
+	hostPort string
+	client   *kubeletutil.KubeletClient
+	hostInfo host.Info
+}
+
+func new(logger *zap.Logger, info host.Info) (*kubeletSummaryProvider, error) {
+	hostIP := os.Getenv("HOST_IP")
```

Comment:
Would it be helpful to check if `HOST_IP` was set correctly or is this checked later in the code?

Maybe we can add
if hostIP == "" {
  // return error or print warning if value is optional
}

Reply:
Before adding this validator, I want to verify a usecase of what happens when `HOST_IP` is not set. Based on results, i can add this additional check. if it fails, then i will add validation but it works then this is not required.